File: easy.server.demo/easy.server.boot.demo/src/main/java/io/github/jinlongliao/easy/server/boot/demo/core/CharsetFilter.java
package io.github.jinlongliao.easy.server.boot.demo.core;

import io.github.jinlongliao.easy.server.servlet.BaseHttpFilter;
import jakarta.servlet.http.HttpServletRequest;
import jakarta.servlet.http.HttpServletResponse;
import org.springframework.stereotype.Component;

import jakarta.servlet.FilterChain;
import jakarta.servlet.ServletException;
import jakarta.servlet.ServletRequest;
import jakarta.servlet.ServletResponse;

import java.io.IOException;

/**
 * @author: liaojinlong
 * @date: 2022-06-28 16:33
 */
@Component
public class CharsetFilter extends BaseHttpFilter {
    @Override
    public String[] supportPath() {
        return new String[]{"/*"};
    }

    @Override
    public boolean doLogicFilter(HttpServletRequest request, HttpServletResponse response) throws IOException, ServletException {
        request.setCharacterEncoding("UTF-8");
        return true;
    }

    @Override
    public boolean isAsync() {
        return true;
    }
}

File: easy.server.demo/easy.server.kotlin.demo/src/main/kotlin/io/github/jinlongliao/easy/server/kotlin/demo/core/CharsetFilter.kt
package io.github.jinlongliao.easy.server.kotlin.demo.core

import io.github.jinlongliao.easy.server.servlet.BaseHttpFilter
import jakarta.servlet.ServletException
import jakarta.servlet.http.HttpServletRequest
import jakarta.servlet.http.HttpServletResponse
import org.springframework.stereotype.Component
import java.io.IOException

/**
 * @author: liaojinlong
 * @date: 2022-06-28 16:33
 */
@Component
class CharsetFilter : BaseHttpFilter() {
    override fun supportPath(): Array<String> {
        return arrayOf("/*")
    }

    @Throws(IOException::class, ServletException::class)
    override fun doLogicFilter(request: HttpServletRequest, response: HttpServletResponse): Boolean {
        request.characterEncoding = "UTF-8"
        return true
    }

    override fun isAsync(): Boolean {
        return true
    }
}
